client/src/modules/cohort/index.jsx

```diff
@@ -122,6 +104,14 @@ class Cohort extends PureComponent {
     return cohortDetails && cohortDetails.isAdmin;
   };
 
+  handleFormDialogVisibility = () => {
```

Comment:
Consider only (age-old conflict): switching visibility with one method or use two methods. In my opinion the first (one method) approach is less intuitive, less self-descriptive, and additionally you have to always destructure the state....